File: net/loveruby/cflat/compiler/Options.java
package net.loveruby.cflat.compiler;

import java.util.List;

public class Options {
    static Options parse(String[] args) {
        Options opts = new Options();
        opts.parseArgs(args);
        return opts;
    }

    private CompilerMode mode;
    // private Platform platform = new X86Linux();
    private String outputFileName;
    // private boolean verbose = false;
    // private LibraryLoader loader = new LibraryLoader();
    private List<SourceFile> sourceFiles;

    private void parseArgs(String[] args) {
        // TODO:
    }

    List<SourceFile> sourceFiles() {
        return sourceFiles;
    }

    String asmFileNameOf(SourceFile src) {
        if (outputFileName != null && mode == CompilerMode.Compile) {
            return outputFileName;
        }
        return src.asmFileName();
    }

    String objFileNameOf(SourceFile src) {
        if (outputFileName != null && mode == CompilerMode.Assemble) {
            return outputFileName;
        }
        return src.objFileName();
    }
}

File: app/src/main/kotlin/com/ksys/cflat/compiler/Options.kt
package com.ksys.cflat.compiler

class Options {
    var mode: CompilerMode? = null
    // vbar platform = X86Linux();
    var outputFileName = ""
    // var verbose = false;
    // var loader = LibraryLoader();
    var sourceFiles = listOf<SourceFile>()

    companion object {
        fun parse(args: List<String>): Options {
            var opts = Options()
            opts.parseArgs(args)
            return opts
        }
    }

    private fun parseArgs(args: List<String>) {
        // TODO:
    }

    fun asmFileNameOf(src: SourceFile): String {
        if (outputFileName != null && mode == CompilerMode.Compile) {
            return outputFileName
        }
        return src.asmFileName()
    }

    fun objFileNameOf(src: SourceFile): String {
        if (outputFileName != null && mode == CompilerMode.Assemble) {
            return outputFileName
        }
        return src.objFileName()
    }
}
